tests/prow_ci.sh

```diff
@@ -30,3 +30,147 @@ override_rosacli_build () {
+    echo "[CI] Login into ${OCM_LOGIN_ENV}"
+    rosa login --env "${OCM_LOGIN_ENV}" --token "${OCM_TOKEN}"
+    ocm login --url "${OCM_LOGIN_ENV}" --token "${OCM_TOKEN}"
+  else
+    echo "Cannot login! You need to specify the offline token OCM_TOKEN!"
+    exit 1
+  fi
+}
+
+# generate_label_filter_switch is used to generate label-filter for the test run
+# IMPORTANCE is step env for job running
+# LABEL_FILTER is global env setting but it can be override by parameter
+# if there is parameter when call this function, it will override
+# Need to clarify the LABEL_FILTER_SWITCH in steps script 
+generate_label_filter_switch () {
+  label_filter=${LABEL_FILTER}
+  if [ ! -z "$1" ]; then
+    echo "[CI] Got parameter label filter: $1 . Override the global setting now."
+    label_filter=$1
+  fi
+  if [[ ! -z "$IMPORTANCE" ]]; then
+    label_filter="$label_filter&&${IMPORTANCE}" 
+  fi
+  LABEL_FILTER_SWITCH="--ginkgo.label-filter '${label_filter}'"
+}
+
+# generate_junit is used to generate junit file path
+# RUN_TIME needs to be set in the step or job level
+# SHARED_DIR is global env for the CI run, no need to set
+# TEST_PROFILE is optional in job level
+# JUNIT_XML need to be clarified in the step definition
+# JUNIT_TEMP_DIR need to be clarified in the step definition
+# there is usage accepts a parameter to override the default usage
+generate_junit () {
+  usage="e2e"
+  if [ ! -z $1 ];then
+    usage="$1"
+  fi
+  JUNIT_TEMP_DIR=$(mktemp -d)
+  junit_file_name="rosa-$usage"
+  if [ ! -z "${TEST_PROFILE}" ]; then
+    junit_file_name="${junit_file_name}-${TEST_PROFILE}"
+  fi
+  if [ ! -z "${RUN_TIME}" ]; then
+    junit_file_name="${junit_file_name}-${RUN_TIME}"
+  fi
+
+  JUNIT_XML="${JUNIT_TEMP_DIR}/${junit_file_name}.xml"
+
+  echo "[CI] the junit temp dir is $JUNIT_TEMP_DIR and junit file will be $JUNIT_XML" 
+}
+
+# extract_existing_junit is used to extract existing junit file from previous steps
+# JUNIT_TEMP_DIR should be clarified in step and generated by generate_junit
+extract_existing_junit () {
+  if [ -z "$JUNIT_TEMP_DIR" ]; then
+    echo "JUNIT_TEMP_DIR is empty, please clarify it and call generate_junit to generate"
+    exit 1
+  fi
+  if [[ -f "${SHARED_DIR}/junit.tar.gz" ]]; then
+      echo "[CI] extract existing junit file package"
+      tar -xvf "${SHARED_DIR}/junit.tar.gz" -C $JUNIT_TEMP_DIR
+  fi
+}
+
+# generate_running_cmd is used to generate rosatest running command
+# TEST_TIMEOUT is step env variable used to define the timeout of the test run
+# JUNIT_XML should be generated by function generate_junit
+generate_running_cmd () {
+  if [ -z $JUNIT_XML ]; then
+    echo "JUNIT_XML is empty, please define it and call generate_junit to generate the value"
+    exit 1
+  fi [ -z $TEST_TIMEOUT ]; then
+    # set a default value in case it is empty
+    TEST_TIMEOUT="4h"
+  fi
+  cmd="rosatest --ginkgo.v --ginkgo.no-color \
+  --ginkgo.timeout ${TEST_TIMEOUT} \
+  --ginkgo.junit-report $JUNIT_XML \
```

Comment:
Are you sure to run all of the test cases if there is no `FOCUS`?

Reply:
There is  ${LABEL_FILTER_SWITCH} controlled the the cases need to run.